scala/src/com/google/idea/blaze/scala/run/producers/ScalaTestContextProvider.java

```diff
@@ -76,32 +92,83 @@ private static ScClass getTestClass(ConfigurationContext context) {
       // handled by a different producer
       return null;
     }
-    return getTestClass(location);
+    ScalaTestConfigurationProducer scalaTestConfigurationProducer = ScalaTestConfigurationProducer.instance();
+    return scalaTestConfigurationProducer.getTestClassWithTestName(location)
+        .map(classWithTestName -> new ClassAndTestName(classWithTestName.testClass(), classWithTestName.testName().getOrElse(() -> null)))
+        .getOrElse(() -> null);
   }
 
-  @Nullable
-  private static ScClass getTestClass(Location<?> location) {
-    PsiElement element = location.getPsiElement();
-    ScClass testClass;
-    if (element instanceof ScClass) {
-      testClass = (ScClass) element;
-    } else {
-      testClass = PsiTreeUtil.getParentOfType(element, ScClass.class);
+  public static class ScalatestTestSelectorFlagsModification implements TestContext.BlazeFlagsModification {
+    @Nullable
+    private final String testName;
+
+    private final String testClassSelectorFlag;
+    private final String testClassFlag;
+    @Nullable
+    private final String testNameSelectorFlag;
+    @Nullable
+    private final String testNameFlag;
+
+    public ScalatestTestSelectorFlagsModification(String testClassFqn, @Nullable String testName) {
+      this.testName = testName;
+      testClassSelectorFlag = BlazeFlags.TEST_ARG + "-s";
+      testClassFlag = BlazeFlags.TEST_ARG + testClassFqn;
+      testNameSelectorFlag = BlazeFlags.TEST_ARG + "-t";
```

Comment:
To better understand what I'm missing in [this comment](https://github.com/bazelbuild/intellij/pull/3744/files/ce9cc2f4f3664ab2161c57edec9a27abba3b2fae#r948508380), why the `testNameSelectorFlag` is not added only if `testName` is not null, i.e. move this line inside the next if?

Reply:
This isn't adding the flag, it's just setting a constant field in this object for later reference. The flags are set here 

https://github.com/bazelbuild/intellij/pull/3744/files/ce9cc2f4f3664ab2161c57edec9a27abba3b2fae#diff-f78adffc6524eb9e6e20bacab71d8bbea8b46ed0d16b60b421ecd064d3f2a122R159-R166

I could make the two fields holding the selector variables static, that might make this clearer?

Reply:
Made the two fields static.

Reply:
Made the two variables static